src/aiida/cmdline/params/types/strings.py

```diff
@@ -20,19 +24,17 @@ class NonEmptyStringParamType(StringParamType):
 
     name = 'nonemptystring'
 
-    def convert(self, value, param, ctx):
-        newval = super().convert(value, param, ctx)
-
-        # Note: Valid :py:class:`click.ParamType`s need to pass through None unchanged
```

Comment:
This was perhaps true in click 7.x but doesn't seem to be true anymore; see docstring of `ParamType.convert()` method:

https://github.com/pallets/click/blob/2a0e3ba907927ade6951d5732b775f11b54cb766/src/click/types.py#L105

And also the code here that calls the convert method only when the value is not None.
https://github.com/pallets/click/blob/2a0e3ba907927ade6951d5732b775f11b54cb766/src/click/types.py#L89